### PR TITLE
[DistSQL] Support create shadow algorithm statement.

### PR DESCRIPTION
**Pull request by lanchengx**
Changes proposed in this pull request:
- Support create shadow algorithm statement.
- Add create shadow algorithm statement test.
- Update create shadow algorithm statement doc.

### CODE REVIEW

**Comment by lanchengx** [comment]
@RaigorJiang  could you have a review ?

File: docs/document/content/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-shadow-rule.en.md [hunk 1]
```diff
@@ -33,7 +35,7 @@ algorithmProperty: key=value
 - `resourceMapping` specifies the mapping relationship between the source database and the shadow library. You need to use the `resource` managed by RDL, please refer to [resource](https:shardingsphere.apache.orgdocumentcurrentcnfeaturesdist-sqlsyntaxrdlrdl-resource)
 - `shadowAlgorithm` can act on multiple `shadowTableRule` at the same time
 - If `algorithmName` is not specified, it will be automatically generated according to `ruleName`, `tableName` and `shadowAlgorithmType`
-- `shadowAlgorithmType` currently supports `COLUMN_REGEX_MATCH` and `SIMPLE_NOTE`
+- `shadowAlgorithmType` currently supports `VALUE_MATCH` and `REGEX_MATCH` and `SIMPLE_HINT`
```
Comment (RaigorJiang) [inline docs/document/content/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-shadow-rule.en.md:38]: supports `VALUE_MATCH`, `REGEX_MATCH` and `SIMPLE_HINT`

File: docs/document/content/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-shadow-rule.cn.md [hunk 2]
```diff
@@ -33,7 +35,7 @@ algorithmProperty: key=value
 - `resourceMapping` 指定源数据库和影子库的映射关系，需使用 RDL 管理的 `resource` ，请参考 [数据源资源](https://shardingsphere.apache.org/document/current/cn/features/dist-sql/syntax/rdl/rdl-resource/)
 - `shadowAlgorithm` 可同时作用于多个 `shadowTableRule`
 - `algorithmName` 未指定时会根据 `ruleName`、`tableName` 和 `shadowAlgorithmType` 自动生成
-- `shadowAlgorithmType` 目前支持 `COLUMN_REGEX_MATCH` 和 `SIMPLE_NOTE`
+- `shadowAlgorithmType` 目前支持 `VALUE_MATCH` 和 `REGEX_MATCH` 以及 `SIMPLE_HINT`
```
Comment (RaigorJiang) [inline docs/document/content/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-shadow-rule.cn.md:38]: 支持 `VALUE_MATCH`、`REGEX_MATCH` 和 `SIMPLE_HINT`

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
@lanchengx Some suggestions for documents.

**Comment by lanchengx** [comment]
@RaigorJiang Modified

**Comment by RaigorJiang** [comment]
@lanchengx Thank you.

**Pull request merged**